src/main/java/com/pharmacy/Pharmacy/mappers/PharmacyBranchMapper.java

```diff
@@ -0,0 +1,41 @@
+package com.pharmacy.Pharmacy.mappers;
+
+import com.pharmacy.Pharmacy.dto.PharmacyBranchDTO;
+import com.pharmacy.Pharmacy.model.PharmacyBranch;
+import com.pharmacy.Pharmacy.service.PharmacyBranchService;
+import com.pharmacy.Pharmacy.service.PharmacyService;
+import org.mapstruct.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Lazy;
+
+@Mapper(componentModel = "spring", injectionStrategy = InjectionStrategy.CONSTRUCTOR)
+public abstract class PharmacyBranchMapper {
+    @Autowired
+    PharmacyService pharmacyService;
+    @Autowired
+    @Lazy
```

Comment:
Dlaczego tutaj ustawiłeś @Lazy ?

Reply:
żeby uniknać circular dependency - podpowiedz znalazlem tutaj https://www.baeldung.com/circular-dependencies-in-spring. Chodzi o to , że zarówno ten mapper jak i kontroler maja zaleznosci do PharmacyBranchService i bez lazy nie wiadomo która zależność ma być wstrzyknięta pierwsza. Jest to jeden z tematów które chcialem przedyskutować, bo może to zły design, że maper woła ten serwis